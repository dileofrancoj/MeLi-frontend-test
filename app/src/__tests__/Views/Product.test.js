import React from "react";
import { shallow } from "enzyme";
import ProductsView from "./../../Views/Products";
import { BrowserRouter as Router } from "react-router-dom";
const product = {
  id: 1,
  title: "titulo",
  price: {
    currency: "ARS",
    amount: 2000,
    decimals: 2,
  },
  picture: "picture.png",
  condition: "new",
  free_shipping: true,
  sold_quantity: 300,
  description: "descripcion de prueba",
  categories: ["producto", "prueba"],
};

describe("<ProductView /> test", () => {
  const wrapper = shallow(
    <Router>
      <ProductsView product={product} />
    </Router>
  );
  test("Should render a ProductView", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
