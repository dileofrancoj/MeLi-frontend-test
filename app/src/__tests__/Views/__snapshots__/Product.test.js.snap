// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductView /> test Should render a ProductView 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProductsView
    product={
      Object {
        "categories": Array [
          "producto",
          "prueba",
        ],
        "condition": "new",
        "description": "descripcion de prueba",
        "free_shipping": true,
        "id": 1,
        "picture": "picture.png",
        "price": Object {
          "amount": 2000,
          "currency": "ARS",
          "decimals": 2,
        },
        "sold_quantity": 300,
        "title": "titulo",
      }
    }
  />
</Router>
`;
